.gas-snapshot

```diff
@@ -19,48 +19,48 @@ Certificate_supportsInterface:test() (gas: 5159)
-Checkout_buyingFromSingleSupplierWithCustomFee:test() (gas: 547923)
-Checkout_buyingFromTenRemovals:test() (gas: 1976973)
-Checkout_buyingFromTenRemovals_singleSupplier:test() (gas: 1965977)
-Checkout_buyingFromTenRemovals_singleSupplier_byApproval:test() (gas: 1963977)
-Checkout_buyingFromTenRemovals_singleSupplier_withoutFee:test() (gas: 1708460)
-Checkout_buyingFromTenRemovals_withoutFee:test() (gas: 1719464)
-Checkout_buyingFromTenSuppliers:test() (gas: 2855260)
-Checkout_buyingWithAlternateERC20:test() (gas: 544721)
-Checkout_buyingWithAlternateERC20_floatingPointPriceMultiple:test() (gas: 544721)
-Checkout_buyingWithCustomFee:test() (gas: 547446)
-Checkout_swapWithDifferentPermitSignerAndMsgSender:test() (gas: 726890)
+Checkout_buyingFromOneRemoval:test() (gas: 726000)
+Checkout_buyingFromOneRemoval_byApproval:test() (gas: 684424)
+Checkout_buyingFromSingleSupplierWithCustomFeeAndPrice:test() (gas: 553566)
+Checkout_buyingFromTenRemovals:test() (gas: 1975477)
+Checkout_buyingFromTenRemovals_singleSupplier:test() (gas: 1964590)
+Checkout_buyingFromTenRemovals_singleSupplier_byApproval:test() (gas: 1962590)
+Checkout_buyingFromTenRemovals_singleSupplier_withoutFee:test() (gas: 1707603)
+Checkout_buyingFromTenRemovals_withoutFee:test() (gas: 1718584)
+Checkout_buyingFromTenSuppliers:test() (gas: 2853764)
+Checkout_buyingWithAlternateERC20:test() (gas: 544459)
+Checkout_buyingWithAlternateERC20_floatingPointPriceMultiple:test() (gas: 544459)
+Checkout_buyingWithCustomFeeAndPrice:test() (gas: 563667)
+Checkout_swapWithDifferentPermitSignerAndMsgSender:test() (gas: 726450)
 LockedNORILib_availableAmount:test() (gas: 12371)
 LockedNORITest:testBatchGrantCreation() (gas: 705179)
 LockedNORITest:testNormalWithdrawal() (gas: 1102743)
 LockedNORITest:testReentryTokensReceived() (gas: 1102887)
 LockedNORITest:testReentryTokensToSend() (gas: 1104432)
 LockedNORITest:testTokensReceivedReverts() (gas: 69026)
-MarketGasExplorer:test_0_bestPossibleCaseOneRemoval() (gas: 29784823)
-MarketGasExplorer:test_1_mostRealisticSeveralSuppliersModerateAmounts() (gas: 54632596)
-MarketGasExplorer:test_2_uniqueSuppliersBulkAmounts() (gas: 77978844)
-MarketGasExplorer:test_3_uniqueSuppliersVariableAmounts() (gas: 1517539092)
-MarketGasExplorer:test_4_singleSupplierGranularAmounts() (gas: 3629646788)
-MarketGasExplorer:test_5_uniqueSuppliersGranularAmounts() (gas: 5791445583)
-MarketInvariantTest:invariant_callSummary() (runs: 400, calls: 6000, reverts: 4273)
-MarketInvariantTest:invariant_sumOfPurchaseAmounts() (runs: 400, calls: 6000, reverts: 4273)
-MarketSupplierSelectionNotUsingUpSuppliersLastRemoval:test() (gas: 922133)
-Market_ALLOWLIST_ROLE:test() (gas: 12799)
-Market_USDC_swap_respects_decimal_mismatch:test() (gas: 716909)
-Market_USDC_swap_withholds_restricted_nori:test() (gas: 892028)
+MarketGasExplorer:test_0_bestPossibleCaseOneRemoval() (gas: 29769122)
+MarketGasExplorer:test_1_mostRealisticSeveralSuppliersModerateAmounts() (gas: 49820714)
```

Comment:
dude.  I have no idea.  there are small changes and improvements all over the place, including for tests that should not have been affected by these changes (since we only made changes to two specific endpoints that only have one test each) 

this makes me think that gas costs are affected by the overall layout of the contract's bytecode, which will have changed a little bit by the change in the implementation of those two endpoints.  that's all I can guess.  maybe chatGPT will know more

Reply:
1) what command did you run to generate the report?
2) Is the last point in this [readme](https://github.com/Rubilmax/foundry-gas-diff#%EF%B8%8F-known-limitations) relevant? I dont think it is, but I'm not positive.

Reply:
I think you're right about the layout. See [this article](https://medium.com/joyso/solidity-how-does-function-name-affect-gas-consumption-in-smart-contract-47d270d8ac92)

Reply:
i'm not sure.... this sentence from the last point in the readme is confusing to me "Average & median gas costs for each function is estimated based on the test suite, which means they are easily impacted by small changes in the tests."  (it's just so abstract)

I just used `forge snapshot` to generate the report

Reply:
I think you might have some compiler options set in your env. I get a different report on this branch when running `yarn snapshot`

Reply:
![Screenshot 2023-08-10 at 1 12 28 PM](https://github.com/nori-dot-eco/contracts/assets/24928141/175514ae-7a9f-45b8-bdd2-43ff97f350bf)
hmm, so I have this in my .env currently -- is this different from what you're working with? (presumably I should be running with the production values for this report?)